Samples/Swift/DaysUntilBirthday/Shared/Services/VerificationLoader.swift

```diff
@@ -14,34 +14,97 @@
  * limitations under the License.
  */
 
+import Combine
 import Foundation
 import GoogleSignIn
 
 #if os(iOS)
 
 /// An observable class for verifying via Google.
 final class VerificationLoader: ObservableObject {
+  /// The `Verification` object of the current user holding the age verification signal.
+  /// - note: Changes to this property will be published to observers.
+  @Published private(set) var verification: Verification?
+
   private var verifiedAgeViewModel: VerifiedAgeViewModel
+  private let baseUrlString = "https://verifywithgoogle.googleapis.com/v1/ageVerification"
+
+  private lazy var components: URLComponents? = {
+    var comps = URLComponents(string: baseUrlString)
+    return comps
+  }()
+
+  private lazy var request: URLRequest? = {
+    guard let components = components, let url = components.url else {
+      return nil
+    }
+    return URLRequest(url: url)
+  }()
 
   /// Creates an instance of this loader.
-  /// - parameter verifiedAgeViewModel: The view model to use to set verification status on.
+  /// - parameter verifiedAgeViewModel: The view model to use to set age verification signal on.
   init(verifiedViewAgeModel: VerifiedAgeViewModel) {
     self.verifiedAgeViewModel = verifiedViewAgeModel
   }
 
+  private func createSession(verifyResult: GIDVerifiedAccountDetailResult,
+                             completion: @escaping (Result<URLSession, Error>) -> Void) {
+    guard let token = verifyResult.accessTokenString else {
+      completion(.failure(.couldNotCreateURLSession))
+      return
+    }
+    let configuration = URLSessionConfiguration.default
+    configuration.httpAdditionalHeaders = [
+      "Authorization": "Bearer \(token)"
+    ]
+    let session = URLSession(configuration: configuration)
+    completion(.success(session))
+  }
+
+  /// Creates a `Publisher` to fetch a user's `Verification` holding their age verification signal.
+  /// - parameter completion: A closure passing back the `AnyPublisher<Verification, Error>`
+  /// upon success.
+  func verificationPublisher(verifyResult: GIDVerifiedAccountDetailResult,
+                             completion: @escaping (AnyPublisher<Verification, Error>) -> Void) {
+    createSession(verifyResult: verifyResult) { [weak self] result in
+      switch result {
+      case .success(let urlSession):
+        guard let request = self?.request else {
+          return completion(Fail(error:.couldNotCreateURLRequest).eraseToAnyPublisher())
+        }
+        let verificationPublisher = urlSession.dataTaskPublisher(for: request)
+          .tryMap { data, error -> Verification in
+            let decoder = JSONDecoder()
+            let verificationResponse = try decoder.decode(VerificationResponse.self, from: data)
+            return verificationResponse.firstVerification
```

Comment:
What happens if the init fails on 78? Maybe that is caught by the `mapError` below? If so, then I think a `throw`ing initializer is good and you can disregard my comment above.

Reply:
Tried this out and can confirm if the init fails on 78 then this is caught by the `mapError`.